subsys/net/lib/coap/coap_client.c

```diff
@@ -820,8 +830,24 @@ static int handle_response(struct coap_client *client, const struct coap_packet
 void coap_client_cancel_requests(struct coap_client *client)
 {
 	for (int i = 0; i < ARRAY_SIZE(client->requests); i++) {
-		client->requests[i].request_ongoing = false;
+		if (client->requests[i].request_ongoing == true) {
+			LOG_DBG("Cancelling request %d", i);
+
+			(void)k_mutex_lock(&client->send_mutex, K_NO_WAIT);
```

Comment:
Not sure I understand this part, what if the mutex was locked already, why are we just ignoring this fact?

Reply:
Removed. I think locking isn't necessary here anyway.

Reply:
I also removed the code that touches the pending list. It is enough to just mark the request as not ongoing.